### PR TITLE
chore: drop support for node v4 and v6

### PR DESCRIPTION
**Pull request by vieiralucas**
This should enable us to merge #1259 and #1260

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,8 +3,6 @@ language: node_js
 addons:
   sauce_connect: true
 node_js:
- - 4 # to be removed 2018-04-30
- - 6 # to be removed 2019-04-01
  - 8 # to be removed 2019-12-31
  - 9 # to be removed 2018-06-30
```
Comment (astorije) [inline .travis.yml]: Probably worth removing v9 as well (end of life since 2018-06-30), and add v10 and v12 to the list, no?

**Comment by vieiralucas** [comment]
@lucasfcosta v9 is actually over, expired in 2018.
I added v10 and v12 as @austince suggested. Did not added v11 since it will expire in 3 days from now 😄

**Comment by austince** [comment]
Believe you meant to tag @astorije 😄

**Comment by astorije** [comment]
Thank you for this!

**Comment by vieiralucas** [comment]
I'll try to fix the `codecov` issue. I believe that we were relying on the fact that we used to run some code in node 4 for coverage (Checking the existance of Proxy).
Just need to add another test.

**Comment by vieiralucas** [comment]
Now I'm very confused on why this is failing :(

**Comment by astorije** [comment]
I can't explain it either but according to [the logs](https://travis-ci.org/chaijs/chai/jobs/538333358#L3601), it's a syntax error coming from [this line](https://github.com/chaijs/chai/blob/master/test/utilities.js#L599). I don't see anything wrong with it though 😕

**Comment by vieiralucas** [comment]
Hey @lucasfcosta and @astorije 
Just fixed the build, problem was that IE 10 didn't support arrow functions (obviously right? 😄 )

I need @chaijs/chai approval again so we can merge this.

**Comment by vieiralucas** [comment]
Also, I just remembered that we did agree with raising IE bar to 11 (see this https://github.com/chaijs/chai/issues/860#issuecomment-395977926).

We just need to move forward with https://github.com/chaijs/guidelines/pull/3

**Comment by ljharb** [comment]
Heads up that this is a breaking change in a minor version - chai v4.0.0 supported node 4, so all versions of chai 4.x need to continue supporting it.

**Pull request merged**